ethcore/src/client/client.rs

```diff
@@ -892,7 +893,7 @@ impl BlockChainClient for Client {
 			ExecutionError::TransactionMalformed(message)
 		})?;
 		let balance = original_state.balance(&sender);
-		let needed_balance = t.value + t.gas * t.gas_price;
+		let needed_balance = t.value + U256::from(UPPER_CEILING) * t.gas_price;
```

Comment:
Should probably use `saturating_add` (if implemented for U256) to avoid overflows.